### ISSUE TITLE
sql: analyze - reintroduce statictics data back

### ISSUE THREAD

**Issue by tsafin** [tarantool/tarantool#4937]
We used to have dat structures which were used for collection of statistics, i.e. see #2843, or #4069.

We need to reintroduce these data structures, but in disabled form first

**Comment by Korablev77** [comment]
Duplicate https://github.com/tarantool/tarantool/issues/4931 ?

**Comment by tsafin** [comment]
> Duplicate #4931 ?

Nope, 4932 is the wrapping epic, this is subtask for adding data-structures, without activating them in the product.

**Comment by Korablev77** [comment]
Data-structures are still in the code: https://github.com/tarantool/tarantool/blob/master/src/box/index_def.h#L71 https://github.com/tarantool/tarantool/blob/master/src/box/index_def.h#L98

They are not even commented.

**Comment by tsafin** [comment]
As part of this activity, we should not forget to make available SQL runtime counters by default.

Now it's hidden by `VDBE_PROFILE` compile-time define (which enable both counters and timing collection). I believe timings are rather expensive and should not be enabled by default, but counters are cheap, and should be always available for query analysis.

**Comment by tsafin** [comment]
> As part of this activity, we should not forget to make available SQL runtime counters by default.

Created separate ticket for this - #6122
